vars/mavenCanaryRelease.groovy

```diff
@@ -65,4 +70,159 @@ def call(body) {
     }
 
     contentRepository(body);
-  }
+}
+
+def patchFMPVersion() {
+    def pomModel = readMavenPom file: 'pom.xml'
+
+    if (!pomModel.profiles) {
+        return false
+    }
+
+    def updatedPlugin = false
+    for(profile in pomModel.profiles) {
+        if(!profile.id.equalsIgnoreCase("openshift") && !profile.build.plugins) {
```

Comment:
would this work as expected? say profile.id is "x" then first condition becomes true and then checks if the profile has build plugins ... which is wrong. 

the `&&` must be `||` 
IMHO. it is ok to be not DRY for clarity 

e.g.

```groovy
if (!profile.equals("openshift") {
   continue
}

if (!profile.build.plugins) {
  continue
}
```